drivers/console/uart_mcumgr.c

```diff
@@ -211,9 +211,10 @@ static int uart_mcumgr_send_raw(const void *data, int len, void *arg)
 	return 0;
 }
 
-int uart_mcumgr_send(const uint8_t *data, int len)
+int uart_mcumgr_send(const uint8_t *data, int len, enum smp_transport_out_flags flags,
```

Comment:
We need to change the function set here. Of  all the parameters only `data` and `len` should be left and the others should be passed as parameters of some context object.
Additional initialization function should be added for initializing of the context.
There is also advantage of using context object: we may calculate frame boundaries in advance and figure out whether we will be able to fit CRC in the last frame or move it, with one byte, to another.
This should allow us to simplify logic in  `mcumgr_serial_tx_frame`.

Reply:
Can use a struct to pass the information through and then access it but I don't follow the rest, the MTU is already known and accessed in the `.zst_get_mtu` function pointer, the serial processing output functions can be vastly optimised but that is separate from this bug

Reply:
Reworked to use a struct variable